File: Cpp/447_Number_of_Boomerangs.cpp
class Solution {
public:
    int numberOfBoomerangs(vector<vector<int>>& points) {
        int result = 0;
        int n = points.size();
        for(int i=0; i<n; ++i){
            unordered_map<int, int> frequencys;
            for(int j=0; j<n; ++j){
                if(i==j) continue;
                int x = points[i][1]-points[j][1];
                int y = points[i][0]-points[j][0];
                int distance = x*x + y*y;
                frequencys[distance] += 1;
            }
            for(auto frequency : frequencys){
                result += (frequency.second * (frequency.second-1));
            }
        }
        return result;
    }
};

File: Kotlin/447_Number_of_Boomerangs.kt
class Solution {
    fun numberOfBoomerangs(points: Array<IntArray>): Int {
        var result = 0
        for(i in points){
            val frequnecy = HashMap<Int,Int>()
            for(j in points){
                if (j.contentEquals(i)) continue
                
                val x = i[1] - j[1]
                val y = i[0] - j[0]
                val distance = x*x + y*y
                frequnecy.put(distance, frequnecy.getOrDefault(distance, 0) + 1)
            }

            for(it in frequnecy){
                result += it.value * (it.value - 1)
            }
        }
        
        return result
    }
}